starter_code/src/movies.js

```diff
@@ -1,19 +1,157 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies) {
+  let rate = [];
+  movies.forEach(function(element) {
+    rate.push(Number(element.rate));
+  });
 
- 
-// Iteration 2: Drama movies - Get the average of Drama Movies
+  const average =
+    rate.reduce(function(total, amount) {
+      return total + amount;
+    }, 0) / movies.length;
+  return parseFloat(average.toFixed(2));
+}
 
+// Iteration 2: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(dramas) {
+  const filtered = dramas.filter(function(movie) {
+    if (movie.genre.indexOf("Drama") === 0) {
+      return true;
+    }
+  });
+  const dramaRating = ratesAverage(filtered);
+  if (filtered.length === 0) {
+    return 0;
+  }
+  return dramaRating;
+}
 
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
 
+function orderByYear(movies) {
+  let order = movies.sort(function(a, b) {
+    // if year is NOT equal
+    if (a.year !== b.year) {
+      return a.year - b.year; // return year ascending
+    }
+    {
+      if (a.title > b.title) {
+        return 1;
+      }
+
+      if (a.title < b.title) {
+        return -1;
+      }
+    }
+  });
+  console.table(order);
+  return order;
+}
+
+// console.log(movies);
+console.clear();
 
 // Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
 
+function howManyMovies(director) {
+  const filtered = director.filter(function(movie, genre) {
+    if (movie.director.indexOf("Steven Spielberg") === 0) {
+      return true;
+    }
+  });
+  console.table(filtered);
+  return filtered.length;
+}
 
 // Iteration 5: Alphabetic Order - Order by title and print the first 20 titles
```

Comment:
Here i would map to create an array with only movie titles. Then sort them and lastly use the splice method to return only the first 20 items.